src/ui/FieldValue/FieldValue.ts

```diff
@@ -228,8 +237,21 @@ export class FieldValue extends Component {
     this.result = this.result || this.resolveResult();
     Assert.exists(this.result);
 
+    if (this.options.condition != null) {
+      var conditionFunction = new Function('obj', 'with(obj||{}){return ' + this.options.condition + '}');
```

Comment:
This unfortunately won't work with Salesforce locker service, nor with a lot of other system. 

Eval'ing code like this would is forbidden by a lot of system where we have to integrate.

The approach we took for this in the past is to instead use "static" condition as a data attribute to determine if a condition is met or not, instead of any arbitrary JavaScript function.

Eg: 
Instead of : `data-condition="raw.snignoreprice==false"`

you have : `data-field-snignoreprice="false"`

https://docs.coveo.com/en/514/javascript-search-framework/understanding-which-result-template-loads-for-a-result

I think you could reuse the `TemplateConditionEvaluation` module for this, with maybe some adaptation.

Reply:
Got it... It would work in ServiceNow though... I inspired that from the TemplateLoader component.

I agree it would be better to find a LockerService-compliant way. I was debating if we should do this logic in the "intializeComponentInside" method and have a generic condition mechanism for all the components getting instantiated within a result. What do you think?

Reply:
Hum 🤔 Interesting idea.

However, there are some drawbacks. 
It becomes some kind of "abstract" option on each component. And we had problem with these in the past.

One of them is discoverability of the option, either through documentation, interface editor etc.

We have similar problem(s) with the `data-tab` option, for example, where a lot of users/clients never learn that it's available.

So I think I'd prefer the simpler approach of adding the option directly on the component (so that it's explicitly documented in the reference documentation), and instead make this an "option type" in the component options module. 

So that if we want to add it to more components, it would be a matter of simply adding it to the components we need with a one liner, instead of adding it for all of them at once.

Reply:
That's what I figured... If I create an option type, do I have to support it in the IE also?

Reply:
Hi Greg, if it's a new option type that the IEditor does not recognize, we treat it as a simple string option and present an input field. That being said, wouldn't a string option work in this case?

Reply:
I'll update the PR, you'll see.